core/inject.ts

```diff
@@ -46,7 +46,10 @@ export function inject(
     containerElement = container;
   }
   // Verify that the container is in document.
-  if (!document.contains(containerElement)) {
+  if (
+    !document.contains(containerElement) &&
+    document !== containerElement?.ownerDocument
```

Comment:
If the first check returns true, won't the second check always return true as well? In this case, can we just remove the first check?

Reply:
Technically yes that should be the case, but I haven't fully tested that theory with iframes, e.g., if we create containerElement in an iframe will document.contains(containerElement) pass if document is the document for `top` rather than the frame. I think it's less risky to add an additional allowance rather than wholesale replace the original logic in the short term.